contrib/admin/cleanup_aws_resources.py

```diff
@@ -0,0 +1,325 @@
+    return buckets
+
+
+def find_sdb_domains_to_cleanup(include_all, match):
+    sdb_domains = dict()
+    for region in regions:
+        print(f'\n[{region}] SimpleDB Domains:')
+        try:
+            sdb_client = aws.client('sdb', region_name=region)
+            domains_in_region = find_sdb_domains_in_region(sdb_client, include_all, match)
+            new_domains = [b for b in domains_in_region if b not in sdb_domains]
+            print('    ' + '\n    '.join(new_domains))
+            for sdb_domain in new_domains:
+                sdb_domains[sdb_domain] = region
+        except Exception as e:
+            # Occurs with botocore.exceptions.SSLError in regions that don't support SDB.
+            # Don't hard-code supported regions, just in case AWS changes these, in order to avoid blind spots.
+            if 'SSL validation failed' in str(e):
+                print('    SimpleDB is not offered in this region.')
+            else:
+                print(f'    An error occurred in this region: {e}')
+    return sdb_domains
+
+
+def find_iam_roles_to_cleanup(include_all, match):
+    iam_roles = dict()
+    for region in regions:
+        print(f'\n[{region}] IAM Roles:')
+        try:
+            iam_client = aws.client('iam', region_name=region)
+            roles_in_region = find_iam_roles_in_region(iam_client, include_all, match)
+
+            new_roles = [b for b in roles_in_region if b not in iam_roles]
+            print('    ' + '\n    '.join(new_roles))
+            for iam_role in new_roles:
+                iam_roles[iam_role] = region
+        except Exception as e:
+            print(f'    An error occurred in this region: {e}')
+    return iam_roles
+
+
+def find_instance_profile_names_to_cleanup(include_all, match):
+    instance_profiles = dict()
+    for region in regions:
+        print(f'\n[{region}] IAM Instance Profiles:')
+        try:
+            iam_resource = aws.resource('iam', region_name=region)
+            iam_client = aws.client('iam')
+            instance_profiles_in_region = find_instance_profile_names_in_region(iam_client, include_all, match)
+
+            new_instance_profiles = [b for b in instance_profiles_in_region if b not in instance_profiles]
+            for new_profile in new_instance_profiles:
+                print(f'    {new_profile}')
+                associated_roles = [r.name for r in iam_resource.InstanceProfile(new_profile).roles]
+                print('     - Roles:')
+                if associated_roles:
+                    print('       - ' + f'\n       - '.join(associated_roles) + '\n')
+                else:
+                    print('       - No roles found to be associated.\n')
+                instance_profiles[new_profile] = region
+        except Exception as e:
+            print(f'    An error occurred in this region: {e}')
+    return instance_profiles
+
+
+def find_buckets_in_region(s3_resource, include_all, match):
```

Comment:
We have both `find_buckets_to_cleanup` and `find_buckets_in_region`. Deciding whether a bucket that exists look suspiciously test-like and needs cleaning up happens... in `find_buckets_in_region`, confusingly. `find_buckets_to_cleanup` just loops over the regions and prints all the buckets. And there aren't docstrings to explain this.

Reply:
Will do some renaming.